sdv/metadata/table.py

```diff
@@ -553,12 +554,32 @@ def _transform_constraints(self, data, on_missing_column='error'):
 
         return data
 
+    def _validate_data_on_constraints(self, data):
+        """Make sure the given data is valid for the given constraints.
+
+        Args:
+            data (pandas.DataFrame):
+                Table data.
+
+        Returns:
+            bool:
+                Whether or not the table data is valid for the constraints.
+        """
+        is_valid = True
+
+        for constraint in self._constraints:
+            if set(constraint.constraint_columns).issubset(data.columns.values):
+                is_valid = is_valid and constraint.is_valid(data).all()
+
+        return is_valid
+
     def transform(self, data, on_missing_column='error'):
         """Transform the given data.
 
         Args:
             data (pandas.DataFrame):
-                Table data.
+                Table data. If the table data is not valid for the provided constraints,
+                a `ValueError` is raised.
```

Comment:
This could be moved to a `Raises` section below (inside the docstring).
I think the exception type should also change:

```
Raises:
    ConstraintsNotMet:
        If the table data is not valid for the provided constraints.
```